freqtrade/freqai/data_kitchen.py

```diff
@@ -0,0 +1,1215 @@
+            backtest_timerange.startts - backtest_period_days * SECONDS_IN_DAY
+        )
+        start = datetime.datetime.utcfromtimestamp(backtest_timerange.startts)
+        stop = datetime.datetime.utcfromtimestamp(backtest_timerange.stopts)
+        full_timerange = start.strftime("%Y%m%d") + "-" + stop.strftime("%Y%m%d")
+
+        self.full_path = Path(
+            self.config["user_data_dir"] / "models" / f"{self.freqai_config['identifier']}"
+        )
+
+        config_path = Path(self.config["config_files"][0])
+
+        if not self.full_path.is_dir():
+            self.full_path.mkdir(parents=True, exist_ok=True)
+            shutil.copy(
+                config_path.resolve(),
+                Path(self.full_path / config_path.parts[-1]),
+            )
+
+        return full_timerange
+
+    def check_if_model_expired(self, trained_timestamp: int) -> bool:
+        """
+        A model age checker to determine if the model is trustworthy based on user defined
+        `expiration_hours` in the configuration file.
+        :params:
+        trained_timestamp: int = The time of training for the most recent model.
+        :returns:
+        bool = If the model is expired or not.
+        """
+        time = datetime.datetime.now(tz=datetime.timezone.utc).timestamp()
+        elapsed_time = (time - trained_timestamp) / 3600  # hours
+        max_time = self.freqai_config.get("expiration_hours", 0)
+        if max_time > 0:
+            return elapsed_time > max_time
+        else:
+            return False
+
+    def check_if_new_training_required(
+        self, trained_timestamp: int
+    ) -> Tuple[bool, TimeRange, TimeRange]:
+
+        time = datetime.datetime.now(tz=datetime.timezone.utc).timestamp()
+        trained_timerange = TimeRange()
+        data_load_timerange = TimeRange()
+
+        # find the max indicator length required
+        max_timeframe_chars = self.freqai_config["feature_parameters"].get(
+            "include_timeframes"
+        )[-1]
+        max_period = self.freqai_config["feature_parameters"].get(
+            "indicator_max_period_candles", 50
+        )
+        additional_seconds = 0
+        if max_timeframe_chars[-1] == "d":
+            additional_seconds = max_period * SECONDS_IN_DAY * int(max_timeframe_chars[-2])
+        elif max_timeframe_chars[-1] == "h":
+            additional_seconds = max_period * 3600 * int(max_timeframe_chars[-2])
+        elif max_timeframe_chars[-1] == "m":
+            if len(max_timeframe_chars) == 2:
+                additional_seconds = max_period * 60 * int(max_timeframe_chars[-2])
+            elif len(max_timeframe_chars) == 3:
+                additional_seconds = max_period * 60 * int(float(max_timeframe_chars[0:2]))
+            else:
+                logger.warning(
+                    "FreqAI could not detect max timeframe and therefore may not "
+                    "download the proper amount of data for training"
+                )
```

Comment:
I'm not sure what this does - and i don't think it's correct either.

the code is using the "last" value of "include_timeframes" - which is a list.
so the value could be '1h', '15m', or anything else (it's also not guaranteed that the list is sorted properly and the highest value is "highest"), so i can break the logic by using the following list (in this sequence): 

```
            "include_timeframes": [
                "3m",
                "15m",
                "1h",
                "5m"
            ],
```

assuming i understood the intend correctly, i think you should loop this list - and use `timeframe_to_seconds()` to convert it to seconds - picking the max from these - and then do whatever transformation is needed.